File: AspNetMvcUnleashed/0672329980_source/Chapter19/CS/UnleashedBlog/UnleashedBlog.Tests/Factories/ControllerFactory.cs
using UnleashedBlog.Controllers;
using UnleashedBlog.Models;
using UnleashedBlog.Tests.Models;


namespace UnleashedBlog.Tests.Factories
{
    public class ControllerFactory
    {
        private BlogRepositoryBase _blogRepository = new FakeBlogRepository();


        public BlogController GetBlogController()
        {
            return new BlogController(_blogRepository);
        }

        public ArchiveController GetArchiveController()
        {
            return new ArchiveController(_blogRepository);
        }

    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter19/VB/UnleashedBlog/UnleashedBlog.Tests/Factories/ControllerFactory.vb
Public Class ControllerFactory

    Private _blogRepository As BlogRepositoryBase = New FakeBlogRepository()


    Public Function GetBlogController() As BlogController
        Return New BlogController(_blogRepository)
    End Function

    Public Function GetArchiveController() As ArchiveController
        Return New ArchiveController(_blogRepository)
    End Function

End Class
